variables.tf

```diff
@@ -72,6 +78,24 @@ variable "custom_network" {
   }
 }
 
+variable "custom_network_security_group" {
+  type        = string
+  default     = ""
+  description = "The name of the custom Azure Virtual Network security group. Only needed when specifying a custom network and using a NAT gateway."
+  validation {
+    condition = length(var.custom_network_security_group) == 0 || can(
+      regex("^/subscriptions/.*/resourceGroups/.*/providers/Microsoft.Network/networkSecurityGroups/.*$", var.custom_network_security_group)
+    )
+    error_message = "Incorrectly formatted network security group input. Use a full Resource ID in this format: /subscriptions/<subscription_id>/resourceGroups/<rg_id>/providers/Microsoft.Network/networkSecurityGroups/<nsg_id>."
+  }
+  validation {
+    condition     = (length(var.custom_network_security_group) == 0  && length(var.custom_network) == 0) || (
+      length(var.custom_network_security_group) != 0 && length(var.custom_network) != 0 && var.use_nat_gateway == true) || (
+      length(var.custom_network_security_group) == 0 && length(var.custom_network) != 0 && var.use_nat_gateway == false)
+    error_message = "You must specify a custom network security group if using a custom virtual network and a NAT gateway. "
```

Comment:
shouldn't this reflect in the custom_vnet example? Currently, I don't see custom_network_security_group being specified in that example.

Reply:
I left it commented out by default, but it's there around line 60 of `examples/custom-vnet/main.tf` with an explanatory comment.

Reply:
I shared my thoughts on this [here](https://github.com/lacework/terraform-azure-agentless-scanning/pull/43#discussion_r1890741369)